AG-15385 Update annotation options and values

alantreadway opened the pull request:
https://ag-grid.atlassian.net/browse/AG-15385

alantreadway committed:
AG-15385 Update annotation options and values

alantreadway committed:
AG-15385 Add groupPercentage support for bar chart annotations

- Add groupPercentage parameter to annotation values for precise positioning within category bands
- Update band scale conversion logic to handle groupPercentage values outside 0-1 range
- Add comprehensive tests for vertical-line, point, and callout annotations with groupPercentage
- Add visual regression snapshots for grouped bar series annotations

alantreadway committed:
Merge branch 'latest' into ag-15385/annotation-options-updates

alantreadway committed:
AG-15385 - nx format.

manapeirov commented on packages/ag-charts-enterprise/src/features/annotations/utils/values.ts:
```diff
@@ -59,8 +60,7 @@ export function invert(
     const width = scale.bandwidth === 0 ? scale.step : scale.bandwidth ?? 0;
     const bandStart = scale.convert(value);
     const bandEnd = bandStart + width;
-    const position = clampArray(n, scale.range);
```
Is it not possible for n to be outside of the scale range?

alantreadway replied:
See TC1 on the linked JIRA

alantreadway committed:
AG-15385 Adjust groupPercentage valid range to [-1, 2] with 0.5 as center

- Update groupPercentage validation range from [-2, 2] to [-1, 2]
- Modify test values to use -0.5 instead of -2 and -1 instead of -1.5
- Update visual regression snapshots for new positioning
- All annotation tests pass with updated range
